wallet/storage/src/internal/store_tx.rs

```diff
@@ -156,7 +156,7 @@ macro_rules! impl_read_ops {
                     .map(Iterator::collect)
             }
 
-            fn exactly_one_root_key(&self) -> crate::Result<bool> {
+            fn root_keys_exist(&self) -> crate::Result<bool> {
```

Comment:
I would call this `check_root_keys_sanity`, and return a result. Also add documentation explaining what the condition of sanity is + error logs printing this error if it happens (at the caller).